Flaky test: test_dictionaries_update_and_reload/test.py::test_reload_while_loading

_Important: This issue was automatically generated and is used by CI for matching failures. DO NOT modify the body content. DO NOT remove labels._

Test name: test_dictionaries_update_and_reload/test.py::test_reload_while_loading
Failure reason: AssertionError: assert 'FAILED_AND_RELOADING' == 'LOADING'
CI report: [Integration tests (amd_asan, db disk, old analyzer, 5/6)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=93237&sha=0b3866953214f2635c552b72ea065ea4dee9ee84&name_0=PR&name_1=Integration%20tests%20%28amd_asan%2C%20db%20disk%2C%20old%20analyzer%2C%205%2F6%29)

CIDB statistics: [cidb](https://play.clickhouse.com/play?user=play&run=1#V0lUSAogICAgOTAgQVMgaW50ZXJ2YWxfZGF5cwpTRUxFQ1QKICAgIHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBBUyBkYXksCiAgICBjb3VudCgpIEFTIGZhaWx1cmVzLAogICAgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikgQVMgcHJzLAogICAgYW55KHJlcG9ydF91cmwpIEFTIHJlcG9ydF91cmwKRlJPTSBjaGVja3MKV0hFUkUgKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZQogICAgQU5EIHRlc3RfbmFtZSA9ICd0ZXN0X2RpY3Rpb25hcmllc191cGRhdGVfYW5kX3JlbG9hZC90ZXN0LnB5Ojp0ZXN0X3JlbG9hZF93aGlsZV9sb2FkaW5nJwogICAgLS0gQU5EIGNoZWNrX25hbWUgPSAnSW50ZWdyYXRpb24gdGVzdHMgKGFtZF9hc2FuLCBkYiBkaXNrLCBvbGQgYW5hbHl6ZXIsIDUvNiknCiAgICBBTkQgdGVzdF9zdGF0dXMgSU4gKCdGQUlMJywgJ0VSUk9SJykKICAgIEFORCAoKHB1bGxfcmVxdWVzdF9udW1iZXIgPSAwIEFORCBoZWFkX3JlZiA9ICdjaV9kb19ub3RfYmxvY2tfbWVyZ2VfYnlfa25vd25fZmFpbHVyZXMnKSBPUiAocHVsbF9yZXF1ZXN0X251bWJlciAhPSAwIEFORCBiYXNlX3JlZiA9ICdtYXN0ZXInKSkKR1JPVVAgQlkgZGF5Ck9SREVSIEJZIGRheSBERVNDCg==)

Test output:
```
File: test_dictionaries_update_and_reload/test.py:135 - in test_reload_while_loading
    assert get_status(instance, "slow") == "LOADING"
E   AssertionError: assert 'FAILED_AND_RELOADING' == 'LOADING'
E     
E     - LOADING
E     + FAILED_AND_RELOADING
```

--- a/src/Common/ShellCommand.cpp
+++ b/src/Common/ShellCommand.cpp
@@ -97,7 +97,7 @@ ShellCommand::~ShellCommand()
 
 bool ShellCommand::tryWaitProcessWithTimeout(size_t timeout_in_seconds)
 {
-    LOG_TRACE(getLogger(), "Try wait for shell command pid {} with timeout {}", pid, timeout_in_seconds);
+    LOG_TRACE(getLogger(), "Try wait for shell command pid {} with timeout {} (seconds)", pid, timeout_in_seconds);
 
     wait_called = true;
 

--- a/tests/integration/test_dictionaries_update_and_reload/test.py
+++ b/tests/integration/test_dictionaries_update_and_reload/test.py
@@ -105,10 +105,6 @@ def test_reload_while_loading(started_cluster):
             timeout=10,
         )
 
-    # The instance should receive the query
-    query("SYSTEM FLUSH LOGS")
-    assert instance.wait_for_log_line(query_id)
-
     # The dictionary is now loading.
     assert get_status(instance, "slow") == "LOADING"
     start_time, duration = get_loading_start_time(
@@ -129,9 +125,6 @@ def test_reload_while_loading(started_cluster):
     query_id = str(uuid.uuid4())
     with pytest.raises(QueryTimeoutExceedException):
         query("SYSTEM RELOAD DICTIONARY 'slow'", query_id=query_id, timeout=10)
-    # The instance should receive the query
-    query("SYSTEM FLUSH LOGS")
-    assert instance.wait_for_log_line(query_id)
     assert get_status(instance, "slow") == "LOADING"
     prev_start_time, prev_duration = start_time, duration
     start_time, duration = get_loading_start_time(

